src/mean.js

```diff
@@ -0,0 +1,17 @@
+'use strict'
+
+/**
+ * Returns the mean of an array of numbers.
+ *
+ * @param {Array<number>} arr
+ * @returns {number}
+ */
+const mean = (arr) => {
+  const length = arr ? arr.length : 0
+  if (length === 0) {
+    return NaN
+  }
+  return arr.reduce((acc, val) => acc + val, 0) / arr.length
```

Comment:
if one of the values is a string of numbers, a concatenation would happen instead of casting to number but it's interesting that's the same behaviour in lodash as well.